src/OpenCensus.Exporter.ApplicationInsights/Implementation/TraceExporterHandler.cs

```diff
@@ -38,38 +39,158 @@ public void Export(IList<ISpanData> spanDataList)
+                    httpPortAttr,
+                    ref name,
+                    ref resultCode,
+                    ref data,
+                    ref target,
+                    ref type,
+                    ref userAgent);
+
+                // BUILDING resulting telemetry
+                OperationTelemetry result;
+                if (resultKind == SpanKind.Client)
                 {
-                    result = new RequestTelemetry();
+                    var resultD = new DependencyTelemetry();
+                    resultD.ResultCode = resultCode;
+                    resultD.Data = data;
+                    resultD.Target = target;
+                    resultD.Type = type;
+
+                    result = resultD;
                 }
                 else
                 {
-                    result = new DependencyTelemetry();
+                    var resultR = new RequestTelemetry();
+                    resultR.ResponseCode = resultCode;
+                    Uri.TryCreate(data, UriKind.RelativeOrAbsolute, out var url);
+                    resultR.Url = url;
+                    result = resultR;
                 }
 
-                result.Name = span.Name;
-                result.Id = span.Context.SpanId.ToLowerBase16();
-                result.Context.Operation.Id = span.Context.TraceId.ToLowerBase16();
-                result.Context.Operation.ParentId = span.ParentSpanId.ToLowerBase16();
-                var duration = span.StartTimestamp.SubtractTimestamp(span.EndTimestamp);
-                result.Duration = TimeSpan.FromTicks((duration.Seconds * TimeSpan.TicksPerSecond) + (duration.Nanos / 100));
-                result.Success = span.Status.IsOk;
-                ((RequestTelemetry)result).ResponseCode = span.Status.Description ?? span.Status.CanonicalCode.ToString();
+                result.Success = success;
 
-                foreach (var attr in span.Attributes.AttributeMap)
+                result.Timestamp = timestamp;
+                result.Name = name;
+                result.Context.Operation.Id = traceId;
+                result.Context.User.UserAgent = userAgent;
+
+                foreach (var prop in props)
+                {
+                    result.Properties.Add(prop.Key, prop.Value);
+                }
+
+                if (parentId != null)
                 {
-                    var value = attr.Value.Match<string>(
-                        (s) => { return s; },
-                        (b) => { return b.ToString(); },
-                        (l) => { return l.ToString(); },
-                        (d) => { return d.ToString(); },
-                        (obj) => { return obj.ToString(); });
+                    result.Context.Operation.ParentId = string.Concat("|", traceId, ".", parentId, ".");
+                }
+
+                // TODO: I don't understant why this concatanation is required
+                result.Id = string.Concat("|", traceId, ".", spanId, ".");
```

Comment:
this is not required in a bright future, but required now for back-compatibility with AppInsights (assuming AI SDK talk to opencensus). We can talk about it and plan to remove it, but for now this is required.